lib/rubocop/cop/rspec/repeated_example_group_body.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Check for repeated describe and context block body.
+      #
+      # @example
+      #
+      #    # bad
+      #    describe 'cool feature x' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    describe 'cool feature y' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    # good
+      #    describe 'cool feature' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    describe 'another cool feature' do
+      #      it { another_predicate }
+      #    end
+      #
+      #    # bad
+      #    context 'when case x' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    context 'when case y' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    # good
+      #    context 'when case x' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    context 'when case y' do
+      #      it { another_predicate }
+      #    end
+      #
+      class RepeatedExampleGroupBody < Cop
+        MSG = 'Repeated %<group>s block body on line(s) %<loc>s'
+
+        def_node_matcher :several_blocks?, <<-PATTERN
+          (begin (block (send _ {#{ExampleGroups::ALL.node_pattern}} ...) ...)+ )
```

Comment:
It's a nice idea to use `#{RSPEC}` instead of the first `_`, it's making sure the receiver is either `nil?` or `:RSpec`, so only `describe do` and `RSpec.describe do` will trigger, not `helper.describe do`.